### ISSUE TITLE
No PR comments on prereleases?

### ISSUE THREAD

**Issue by mogelbrod** [semantic-release/github#277]
I've got a private NPM (GitHub package repository to be exact) package released via semantic-release with a `beta` prerelease channel: [**release.config.js**](https://gist.github.com/mogelbrod/cbb750eca494f22c3380d44de5529ef1)

Whenever I wish to trigger a beta release (by following [these instructions](https://github.com/semantic-release/semantic-release/blob/master/docs/recipes/pre-releases.md#working-on-a-future-release)) I do the following:

1. `git checkout feat1` - check out the branch (with a corresponding open GitHub PR) I'd like to prerelease
2. `git merge master && git mergetool && git commit` - (optionally) merge with master
3. `git checkout beta && git pull` - fetch the latest `beta` prerelease
4. `git merge feat1 && git mergetool && git commit` - apply feat1, taking care to keep the current`beta` version in package.json
5. `git push origin` - trigger release

The release finishes successfully and publishes the correct beta version to the registry.
However it never seems to post a comment to the open `feat1` PR, instead opting to post comments to merged PRs that have already been released on the master branch (and therefore already have a "released in version X" comment) - similar to [this PR](https://github.com/semantic-release/github/pull/233).

Is this caused by bad configuration, or is there anything else I can do to not have to resort to manually writing "released in X" comments on the PR?

By the looks of https://github.com/semantic-release/github/issues/105#issuecomment-418170667 it sounds like I need to reference the PR number in the merge commit, is that correct?

Thanks in advance!

**Comment by mogelbrod** [comment]
Still very much interested in this 🙏

**Comment by travi** [comment]
i dont think i understand the desired workflow. comments are added to _closed_ PRs and issues so that followers of those threads can be notified that the related changes have been released. therefore, the comments do not apply to _open_ PRs like you seem to be asking for.

in addition, it is expected that the pre-release branch would normally be created from your mainline branch (sounds like this would be `master` in your example). if following that approach, i would not expect step 2 to be necessary. if you are creating the pre-release branch from some other branch and then merging your mainline into it, this would result in the behavior you describe where comments are added to threads that have already been released on the mainline. that is because semantic-release relies heavily on history as defined by git and the merge results in those commits happening _after_ you initiated the pre-release channel. please keep in mind that semantic-release does not officially support gitflow, which it sounds like might be the process you are trying to use. instead, we strongly encourage using github flow or trunk-based development when working on packages.

**Comment by mogelbrod** [comment]
Thanks for the extensive reply, much appreciated!
Am I understanding you correct in that the semantic-release workflow is to
1. Branch off `beta` from `master`
2. Merge PRs that we wish to include in the `beta` pre-release into `beta` (optionally using the Github PR UI after changing the PR base branch to `beta`)
3. semantic-release creates pre-releases for every PR merged into `beta`
4. Once `beta` is ready for a stable release we merge it back into `master`

Should the `beta` branch be re-created from `master` the next time it's time for a `beta` pre-release, or is it typically kept around and new PRs simply merged into it on demand?

**Comment by babblebey** [comment]
> Should the `beta` branch be re-created from `master` the next time it's time for a `beta` pre-release, or is it typically kept around and new PRs simply merged into it on demand?

You can operate this in whatever way that suits you 😉, but it is an known practice to have the `beta` kept around as the branch where all PRs are typically merge into as the beta testing version which keeps it ahead of the `master`/`main` branch which serves as production branch... and as soon as a the current features on the `beta` is seen as stable, you merge it to the `master`/`main`, hence triggering a latest release.

This is the current practice at https://github.com/open-sauced/app 😉 

Hope this help 😁

**Comment by travi** [comment]
> Should the `beta` branch be re-created from `master` the next time it's time for a `beta` pre-release, or is it typically kept around and new PRs simply merged into it on demand?

apologies for the delayed response here. yes, the beta branch is expected to be deleted and recreated when you need to get feedback on another pre-release by your consumers. once you have promoted the change to stable, the next time you chose to make a pre-release should be a new branch from the latest HEAD of your stable release branch. this should not be needed for every release.

> it is an known practice to have the `beta` kept around as the branch where all PRs are typically merge into as the beta testing version which keeps it ahead of the `master`/`main` branch which serves as production branch... and as soon as a the current features on the `beta` is seen as stable, you merge it to the `master`/`main`, hence triggering a latest release.

i want to clarify what might seem confusing. what is being described here sounds like gitflow, which is not officially supported by semantic-release. while it is a popular branching workflow, it promotes working in a style that is not aligned with the continuous delivery approach that semantic-release normally encourages. semantic-release not support a workflow that enables further verification that a release is working as intended. while some do use our pre-release workflow to simulate gitflow, it is not the intended use. pre-releases are for gaining feedback about a proposed or unstable api adjustment from consumers, not to enable testing after releases.

**Comment by mogelbrod** [comment]
Thanks for the clarifications, appreciate it!
